### PR TITLE
Remove fork when redirecting stderr in onnx-mlir 

### PR DESCRIPTION
**Pull request by NathanielMcVicar**
Removes fork from onnx-mlir CL utility, to avoid child processes hanging on exit(0). In general, we don't expect a naked fork() to be safe within LLVM threading (the MLIR pass manager is multithreaded in this case).

Adds a command line option to change the output base filename.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -21,6 +21,9 @@ int main(int argc, char *argv[]) {
       llvm::cl::desc("<input file>"), llvm::cl::init("-"),
       llvm::cl::cat(OnnxMlirOptions));
 
+  llvm::cl::opt<string> outputBaseName("output-base",
```
Comment (tjingrant) [inline src/main.cpp:24]: Can we use `-o` like we do with `gcc` or `clang`?

**Comment by tjingrant** [comment]
@doru1004 can you take a look at this patch?

**Review by caoimhinuibrian — COMMENTED** [review]
LGTM

**Review by doru1004 — APPROVED** [review]
LGTM

**Comment by caoimhinuibrian** [comment]
@jenkins-droid test this please.

**Comment by NathanielMcVicar** [comment]
@jenkins-droid test this please.

**Comment by caoimhinuibrian** [comment]
@jenkins-droid test this please.

**Comment by NathanielMcVicar** [comment]
@caoimhinuibrian I guess jenkins-droid ignores me because I'm out of org?

**Comment by caoimhinuibrian** [comment]
@NathanielMcVicar yes, I think so ... not entirely clear what the criteria are ... @tjingrant can probably enlighten

**Pull request merged**